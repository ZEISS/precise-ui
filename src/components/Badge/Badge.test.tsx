import * as React from 'react';
import * as enzyme from 'enzyme';
import { Badge } from './';
import 'jest-styled-components';

describe('<Badge />', () => {
  it('should render an empty <Badge> component', () => {
    const wrapper = enzyme.shallow(<Badge />);
    expect(wrapper.text()).toBe('<styled.div />');
  });

  it('should render a <Badge> with text', () => {
    const wrapper = enzyme.shallow(<Badge>0</Badge>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Badge> in full mode', () => {
    const wrapper = enzyme.shallow(<Badge fill>12</Badge>);
    expect(wrapper).toMatchSnapshot();
  });
});
