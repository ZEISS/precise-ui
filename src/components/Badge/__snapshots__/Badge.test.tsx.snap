// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render a <Badge> in full mode 1`] = `
<styled.div>
  12
</styled.div>
`;

exports[`<Badge /> should render a <Badge> with text 1`] = `
<styled.div>
  0
</styled.div>
`;
